packages/lime-system/files/usr/lib/lua/lime/system.lua

```diff
@@ -15,12 +15,36 @@ function system.get_hostname()
 end
 
 function system.set_hostname()
-	local hostname = system.get_hostname() 
+	local hostname = system.get_hostname()
 	local uci = config.get_uci_cursor()
 	uci:foreach("system", "system", function(s) uci:set("system", s[".name"], "hostname", hostname) end)
 	uci:save("system")
 end
 
+function system.setup_root_password()
+
+	local policy = config.get("system", "root_password_policy")
+
+	if policy == "DO_NOTHING" then
+		--! nothing...
+	elseif policy == "SET_SECRET" then
+		local secret = config.get("system", "root_password_secret")
+		local current_secret = utils.get_root_secret()
+		if current_secret == nil then
+			error("Can't get root password")
+		end
+		if current_secret ~= secret then
+			utils.set_root_secret(secret)
+		end
+	elseif policy == "RANDOM" then
+		if #utils.get_root_secret() <= 1 then
```

Comment:
`<= 1` it's because first line might be an empty line on `\n` if not set? In that case I think this should be handled directly inside `utils.get_root_secret()` returning `nil` if the first line of `/etc/shadow` is empty/invalid

Reply:
Some possible options are `*` (password never set, and empty) or `!` account locked. For example:
```
root::0:0:99999:7:::
daemon:*:0:0:99999:7:::
ftp:*:0:0:99999:7:::
```
I can add a comment here to clarify this.

Reply:
ok good to comment it ;)